parsec/core/local_storage.py

```diff
@@ -109,13 +113,26 @@ def get_manifest(self, entry_id: EntryID) -> LocalManifest:
         self.local_manifest_cache[entry_id] = manifest
         return manifest
 
-    def set_manifest(self, entry_id: EntryID, manifest: LocalManifest) -> None:
+    def set_manifest(
+        self, entry_id: EntryID, manifest: LocalManifest, cache_only: bool = False
+    ) -> None:
         assert isinstance(entry_id, EntryID)
         assert manifest.author == self.device_id
         self._check_lock_status(entry_id)
+        if not cache_only:
+            raw = local_manifest_serializer.dumps(manifest)
+            self.persistent_storage.set_manifest(entry_id, raw)
+        self.local_manifest_cache[entry_id] = manifest
```

Comment:
Maybe discard the `entry_id` as well:

    self.cache_ahead_of_persistance_ids.discard(entry_id)

Reply:
done